### PR TITLE
PyGrid Gateway Persistence

### PR DESCRIPTION
**Pull request by IonesioJunior**
# Pull Request Template

## Description

Develop Persistence Schema described below.

```
Model
- id: ID
- name: STRING
- version: STRING
- checkpoints: [
  1: ModelCheckpoint
  2: ModelCheckpoint
  3: ModelCheckpoint
  my-favorite: ModelCheckpoint
  5: ModelCheckpoint
  6: ...
]

ModelCheckpoint
- id: ID
- value: TENSOR (value of the model at a given checkpoint)

Plan
- id: ID
- value: STRING (code of plan as list of operations)
- value_ts: STRING (code of plan as TorchScript)

Protocol
- id: ID
- value: STRING (code of protocol as list of operations)
- value_ts: STRING (code of protocol as TorchScript)

Config
- id: ID
- value: DICT

Worker
- id: ID
- format_preference: STRING (either "list" or "ts")
- ping: INT
- avg_upload: INT
- avg_download: INT
- worker_cycles: [WorkerCycle, WorkerCycle, ...]

WorkerCycle
- id: ID
- fl_process: FlProcess
- cycle: Cycle
- worker: Worker
- request_key: TOKEN (unique token that permits downloading specific Plans, Protocols, Configs, and ModelCheckpoints)

Cycle
- id: ID
- start: TIME
- end: TIME
- worker_cycles: [WorkerCycle, WorkerCycle, ...]

FLProcess
- id: ID
- averaging_plan: Plan
- plans: [
  training-plan: Plan
  validation-plan: Plan
]
- protocols: [
  secure-agg-protocol: Protocol
]
- client_config: Config
- server_config: Config
- model: Model
- cycles: [Cycle, Cycle, ...]
```

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
- [ ] New Unit tests added
- [X] Unit tests pass locally with my changes

### CODE REVIEW

**Comment by hericlesme** [comment]
@IonesioJunior, do you intend to run these tests alongside others with pytest? In the CI build, for example.

**Comment by IonesioJunior** [comment]
> @IonesioJunior, do you intend to run these tests alongside others with pytest? In the CI build, for example.

Yup!

**Comment by hericlesme** [comment]
> > @IonesioJunior, do you intend to run these tests alongside others with pytest? In the CI build, for example.
> 
> Yup!

Do the tests work already? If so, it would be a good idea to put them in the `/test` folder, so the CI can get them. :)

**Comment by vvmnnnkv** [comment]
Suggestion: have "alias" field in the model checkpoint. So that checkpoint IDs are always auto-incremented numbers, but you can also assign text alias to any of them.

File: test/test_gateway_sockets.py [hunk 1]
```diff
@@ -42,6 +55,132 @@ class GatewaySocketsTest(aiounittest.AsyncTestCase):
         response = await send_ws_message({"type": "socket-ping", "data": {}})
         self.assertEqual(response, {"alive": "True"})
 
+    async def test_fl_process(self):
```
Comment (IonesioJunior) [inline test/test_gateway_sockets.py]: @cereallarceny , @Prtfw , @hericlesme  Please, take a look at this method.
This unit test method shows how to use the WebSocket interface to host a new training process ( sending serialized models, cycles and protocols), perform a simple authentication and request to join in a new FL cycle

Comment (cereallarceny) [reply test/test_gateway_sockets.py]: LGTM!  I will say that technically the `host_federated_training` method exists on PySyft, not PyGrid. But for testing purposes... that's okay. :)

**Pull request merged**